(Publish 3/15 in AM) "What's new" GPT3 launch post (#12068)
newrelic/docs-website: src/content/whats-new/2023/03/whats-new-3-14-new-relic-openai-integration.md

---

> title: 'Monitor OpenAI’s GPT-3 Application Usage in New Relic'

Reviewer: ```suggestion
title: 'Monitor OpenAI’s GPT Application Usage in New Relic'
```

> summary: 'Monitor GPT-3 and OpenAI applications with just two lines of code using New Relic to track the cost, usage, and performance of your models in real-time.'

Reviewer: ```suggestion
summary: 'Monitor OpenAI applications running GPT-3 and GPT-4 with just two lines of code using New Relic to track the cost, usage, and performance of your models in real-time.'
```

> summary: 'Monitor OpenAI applications running GPT-3 and GPT-4 with just two lines of code using New Relic to track the cost, usage, and performance of your models in real-time.'

Reviewer: ```suggestion
summary: 'Monitor OpenAI applications running GPT-3, GPT-3.5, and beyond with just two lines of code using New Relic to track the cost, usage, and performance of your models in real-time.'
```

> releaseDate: '2023-03-14'

Reviewer: ```suggestion
releaseDate: '2023-03-15'
```

learnMoreLink: 'https://newrelic.com/instant-observability/openai'
getStartedLink: 'https://github.com/newrelic/nr-openai-observability'
---

> If you're already using, or plan to use, OpenAI's GPT-3 and GPT-3.5 at scale, it's important to monitor metrics like average request time, total requests, and total cost. Doing so can help you ensure that services using ChatGPT, GPT-3, and GPT-3.5 are working as expected, especially when those services are required for important functions like customer service and support.

Reviewer: ```suggestion
If you're already using, or plan to use, OpenAI's GPT-3, GPT-3.5, or GPT-4 at scale, it's important to monitor metrics like average request time, total requests, and total cost. Doing so can help you ensure that OpenAI services like ChatGPT are working as expected, especially when those services are required for important functions like customer service and support.
```

> Companies around the world are embracing GPT-3 to power help desk tickets and live chat logs, develop content and images, and accelerate semantic searches. New Relic monitoring for OpenAI is fast, easy to use, and unlocks real-time metrics that can help engineering teams optimize usage, reduce costs, and achieve better results.

Reviewer: ```suggestion
Companies around the world are embracing GPT-3 and GPT-4 to power help desk tickets and live chat logs, develop content and images, and accelerate semantic searches. New Relic monitoring for OpenAI is fast, easy to use, and unlocks real-time metrics that can help engineering teams optimize usage, reduce costs, and achieve better results.
```

> Companies around the world are embracing GPT-3 and GPT-4 to power help desk tickets and live chat logs, develop content and images, and accelerate semantic searches. New Relic monitoring for OpenAI is fast, easy to use, and unlocks real-time metrics that can help engineering teams optimize usage, reduce costs, and achieve better results.

Reviewer: ```suggestion
Companies around the world are embracing GPT to power help desk tickets and live chat logs, develop content and images, and accelerate semantic searches. New Relic monitoring for OpenAI is fast, easy to use, and unlocks real-time metrics that can help engineering teams optimize usage, reduce costs, and achieve better results.
```

> ## New Relic integrates with GPT-3 and OpenAI to provide the following benefits:

Reviewer: ```suggestion
## New Relic integrates with GPT-3, GPT-4, and OpenAI to provide the following benefits:
```

> ## New Relic integrates with GPT-3, GPT-4, and OpenAI to provide the following benefits:

Reviewer: ```suggestion
## New Relic integrates with OpenAI, GPT-3, and beyond to provide the following benefits:
```

> ![New Relic's OpenAI integration providing a pre-built dashboard for monitoring performance, cost, and usage of GPT-3 applications. ](./images/newrelic_openai_integration_dashboard.png "A screenshot showing the New Relic Instant Observability quickstart dashboard for OpenAI.")

Reviewer: ```suggestion
![New Relic's OpenAI integration providing a pre-built dashboard for monitoring performance, cost, and usage of GPT applications. ](./images/newrelic_openai_integration_dashboard.png "A screenshot showing the New Relic Instant Observability quickstart dashboard for OpenAI.")
```

> * __Get started for free:__ Access to New Relic Instant Observability and our out-of-the-box GPT-3 monitoring solution is the first of its kind, and included at no additional cost to all New Relic full platform users.

Reviewer: ```suggestion
* __Get started for free:__ Access to New Relic Instant Observability and our out-of-the-box GPT monitoring solution is the first of its kind, and included at no additional cost to all New Relic full platform users.
```

> * __Easy installation:__ With just two lines of code, users can import the monitor module from the `nr_openai_monitor` library and automatically generate a dashboard that displays a variety of key GPT-3 performance metrics.

Reviewer: ```suggestion
* __Easy installation:__ With just two lines of code, users can import the monitor module from the `nr_openai_monitor` library and automatically generate a dashboard that displays a variety of key GPT performance metrics.
```

> * __Easy installation:__ With just two lines of code, users can import the monitor module from the `nr_openai_monitor` library and automatically generate a dashboard that displays a variety of key GPT-3 performance metrics.

Reviewer: ```suggestion
* __Easy installation:__ With just two lines of code, users can import the monitor module from the `nr_openai_monitor` library and automatically generate a dashboard that displays a variety of key GPT performance metrics.
```

> * __Monitor cost:__ Usage of OpenAI’s Davinci model costs $0.12 per 1000 tokens, which can add up quickly and make it difficult to operate at scale. New Relic provides engineering teams with real-time cost tracking of their GPT-3 usage.

Reviewer: ```suggestion
* __Monitor cost:__ Usage of OpenAI’s Davinci model costs $0.12 per 1000 tokens, which can add up quickly and make it difficult to operate at scale. New Relic provides engineering teams with real-time cost tracking of their GPT usage.
```

> * __Optimize performance:__ New Relic gives engineering teams insight into the average response time and other key performance metrics around GPT-3 requests, allowing engineers to optimize usage and ensure the best possible response times.

Reviewer: ```suggestion
* __Optimize performance:__ New Relic gives engineering teams insight into the average response time and other key performance metrics around GPT requests, allowing engineers to optimize usage and ensure the best possible response times.
```

> * __Analysis prompts and response:__ New Relic provides valuable information about the usage, speed, and effectiveness of GPT-3 to help engineering teams achieve better results from their ML models.

Reviewer: ```suggestion
* __Analyze prompts and responses:__ New Relic provides valuable information about the usage, speed, and effectiveness of GPT to help engineering teams achieve better results from their ML models.
```

## How to get started

> With just two lines of code, simply import the monitor module from the `nr_openai_monitor` library and automatically generate a dashboard that displays a variety of key GPT-3 and GPT-3.5 performance metrics such as cost, requests, average response time, and average tokens per request.

Reviewer: ```suggestion
With just two lines of code, simply import the monitor module from the `nr_openai_monitor` library and automatically generate a dashboard that displays a variety of key GPT-3 and GPT-4 performance metrics such as cost, requests, average response time, and average tokens per request.
```

> With just two lines of code, simply import the monitor module from the `nr_openai_monitor` library and automatically generate a dashboard that displays a variety of key GPT-3 and GPT-4 performance metrics such as cost, requests, average response time, and average tokens per request.

Reviewer: ```suggestion
With just two lines of code, simply import the monitor module from the `nr_openai_monitor` library and automatically generate a dashboard that displays a variety of key GPT-3 and GPT-3.5 performance metrics such as cost, requests, average response time, and average tokens per request.
```
